spi: Grab relevant properties from DTS into the operation bitfield

tbursztyka opened the pull request:
duplex and frame_format where recently added and obviously need to be
grabbed from DTS relevantly.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
spi: Grab relevant properties from DTS into the operation bitfield

duplex and frame_format where recently added and obviously need to be
grabbed from DTS relevantly.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

erwango commented on include/drivers/spi.h:
```diff
@@ -297,7 +297,9 @@ struct spi_config {
-		.operation = (operation_),				\
+		.operation = (operation_) |				\
+			DT_PROP_OR(node_id, duplex, 0) |		\
+			DT_PROP_OR(node_id, frame_format, 0),		\
```
Since `frame-format` is defined with `default: 0` in binding, the `_OR` isn't needed.

tbursztyka replied:
ok, thus the error on duplex then. I'll switch back to DT_PROP() and fix the dts bindings for duplex.

Commit:
dts: Set a default value for SPI device's duplex property

Better setting a default value there so DT_PROP() will return something
relevant whether the property is set or not.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>
